crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -613,26 +634,23 @@ impl<'a, Provider: ResolverProvider> Resolver<'a, Provider> {
                     .ok_or(ResolveError::Unregistered)?;
                 self.visited.insert(package_name.clone());
 
+                let empty_version_map = VersionMap::default();
```

Comment:
We need to initialize this for ownership since `version_map` is a reference.

Reply:
Alternatively we could have `CandidateSelector.select` take an `Option<&VersionMap>` but I opted for an empty iterable to avoid nesting in `select` instead. I'd consider revisiting this in the future as well.

Reply:
I _think_ you can do `let empty_version_map;` here and then `empty_version_map = VersionMap::default()` in each branch if you want.

Reply:
Oh that could be nicer.

Reply:
Actually idk if that makes sense we're trying to return `&VersionMap::default()` in the match which is the problem?